kubernetes/template-generator/cli.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+from template_generator import *
+
+
+LEAPP_BASEPATH = "/var/lib/leapp/macrocontainers"
+
+
+def parse_params():
+    def _parse_port(arg):
+        docker_port, sep, container_port = arg.partition(":")
+        if not container_port:
+            container_port = docker_port
+
+        return int(docker_port), int(container_port)
+
+    def _parse_ip(arg):
+        ## TODO add proper parser
+        return arg
+
+    parser = argparse.ArgumentParser(description='Generate kubernetes tempates.')
+    subparser = parser.add_subparsers(dest="generator")
+
+    multi = subparser.add_parser("multivolume", help="test")
+    macro = subparser.add_parser("macroimage", help="test2")
+
+    #selectors = []
+    mandatory_groups = []
+
+    for p in multi, macro:
+        p.add_argument("--tcp", "-t", type=_parse_port, nargs="*", help="Exposed TCP ports")
+        p.add_argument("--ip", "-i", type=_parse_ip, nargs="+", help="IP address on which the ports will be exposed")
+        p.add_argument("--dest", "-e", default=os.getcwd(), help="Generate service template (default: true)")
+
+        mandatory = p.add_argument_group("Mandatory arguments")
+        mandatory.add_argument("--container-name", "-c", help="A name of macrocontainer", required=True)
+        mandatory_groups.extend([mandatory])
+
+        #selector = p.add_argument_group("Template selectors (optional)")
+        #selector.add_argument("--pod", "-p", action="store_true", default=None, help="Generate pod template (default: true)")
+        #selector.add_argument("--service", "-s", action="store_true", default=None, help="Generate service template (default: true)")
+        #selectors.extend([selector])
+
+
+    #multivolume specific
+    mandatory_groups[0].add_argument("--image-url", "-m", help="URL to machine tar image", required=True)
+
+    #macro specific
+    #selectors[1].add_argument("--docker", "-d", action="store_true", default=None, help="Generate dockerfile (default: true)")
+    macro.add_argument("--local-image", "-l", action="store_true", help="Disable pulling images on node (Default: false)", default=False)
+
+    return parser.parse_args()
+
+
+def write_file(path, content):
+    f = open(path, "w")
+    f.write(content)
+    f.close()
+
+def main():
+    params = parse_params()
+
+    path = "{}/{}".format(LEAPP_BASEPATH, params.container_name)
+
+    ## Find release version
+    rfile = open("{}/etc/redhat-release".format(path), "r")
+    ver = re.findall(".*\srelease\s(\d+)(?:\.\d+)?.*", rfile.readline(64))
+    rfile.close()
+
+    if len(ver) and ver[0] == "6":
```

Comment:
again all this can be replaced with https://docs.python.org/2/library/platform.html